### PR TITLE
simplify and fix complete_payment helper and refactor tests

### PR DESCRIPTION
**Pull request by dobertRowneySr**
Addresses:

* #3688 
* #3657 

#### Changes
* implemented royalty fee accounting proposed in #3688 
* `complete_payment` helper function cleaned up
* add test for case described in the security report with proper refactoring

#### Notes
* rebased on `carthage-substrate-v3`

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202132419573087/1202476311049904) by [Unito](https://www.unito.io)

### CODE REVIEW

File: runtime-modules/content/src/tests/mock.rs [hunk 1]
```diff
@@ -536,7 +536,59 @@ impl Default for ExtBuilder {
...
+            min_starting_price: self.min_starting_price,
+            max_starting_price: self.max_starting_price,
+            min_creator_royalty: self.min_creator_royalty,
+            max_creator_royalty: self.max_creator_royalty,
+            min_bid_step: self.min_bid_step,
+            max_bid_step: self.max_bid_step,
+            platform_fee_percentage: self.platform_fee_percentage,
+            auction_starts_at_max_delta: self.auction_starts_at_max_delta,
+            max_auction_whitelist_length: self.max_auction_whitelist_length,
+        }
```
Comment (Lezek123) [inline runtime-modules/content/src/tests/mock.rs]: Probably worth refactoring this into a separate method

File: runtime-modules/content/src/tests/nft/buy_nft.rs [hunk 2]
```diff
@@ -4,62 +4,70 @@ use crate::tests::mock::*;
...
+                VideoId::one(),
+                SECOND_MEMBER_ID,
+                DEFAULT_NFT_PRICE,
+            ));
+
+            // nft_price * min(100%, 100% - platform_fee%) = nft_price - platform_fee
+            assert_eq!(
+                channel_reward_account_balance(1u64),
+                DEFAULT_NFT_PRICE - platform_fee,
+            );
```
Comment (Lezek123) [inline runtime-modules/content/src/tests/nft/buy_nft.rs]: You're only checking one account, but the original issue involved 2 accounts.
I suggest adding a check of `SECOND_MEMBER_ACCOUNT_ID` (- nft_price) and `COLLABORATOR_MEMBER_ID` (no balance change) also

**Review by Lezek123 — COMMENTED** [review]
Looks good, just added 2 minor suggestions

**Comment by Lezek123** [comment]
One more thing - please run `cargo fmt`

File: runtime-modules/content/src/tests/mock.rs (hunk 1 above)
Comment (dobertRowneySr) [reply runtime-modules/content/src/tests/mock.rs]: see 2228e1ed689475a41645a46f0da104a98894cdaf it's similar to the project token pallet now

**Review by Lezek123 — COMMENTED** [review]
LGTM, but the metadata and types need to be updated

**Review by Lezek123 — APPROVED** [review]
LGTM

**Pull request merged**